server/src/main/java/org/opensearch/index/compositeindex/datacube/startree/fileformats/node/FixedLengthStarTreeNode.java

```diff
@@ -271,6 +290,88 @@ private FixedLengthStarTreeNode binarySearchChild(long dimensionValue) throws IO
         return null;
     }
 
+    @Override
+    public void collectChildrenInRange(Long low, Long high, StarTreeNodeCollector collector) throws IOException {
+        if (low <= high) {
+            FixedLengthStarTreeNode lowStarTreeNode = binarySearchChild(low, true, null);
+            if (lowStarTreeNode != null) {
+                FixedLengthStarTreeNode highStarTreeNode = binarySearchChild(high, false, lowStarTreeNode);
+                if (highStarTreeNode != null) {
+                    for (int lowNodeId = lowStarTreeNode.nodeId(); lowNodeId <= highStarTreeNode.nodeId(); ++lowNodeId) {
+                        collector.collectStarNode(new FixedLengthStarTreeNode(in, lowNodeId));
+                    }
+                } else { // Low StarTreeNode is the last default node for that dimension.
+                    collector.collectStarNode(lowStarTreeNode);
+                }
+            }
+        }
+    }
+
+    private FixedLengthStarTreeNode binarySearchChild(long dimensionValue, boolean matchNextHighest, StarTreeNode lastMatchedNode)
+        throws IOException {
+
+        int low = firstChildId;
+        int tempLow = low;
+
+        // if the current node is star node, increment the tempLow to reduce the search space
+        if (matchStarTreeNodeTypeOrNull(new FixedLengthStarTreeNode(in, tempLow), StarTreeNodeType.STAR) != null) {
+            tempLow++;
+        }
+
+        int high = getInt(LAST_CHILD_ID_OFFSET);
+        int tempHigh = high;
+        // if the current node is null node, decrement the tempHigh to reduce the search space
+        if (matchStarTreeNodeTypeOrNull(new FixedLengthStarTreeNode(in, tempHigh), StarTreeNodeType.NULL) != null) {
+            tempHigh--;
+        }
+
+        if (lastMatchedNode instanceof FixedLengthStarTreeNode) {
```

Comment:
same comment as above, we can move up and skip get low

Reply:
Since, I need star and null node Ids we would need starNodeId irrespective of lastMatchedNode exists or not.